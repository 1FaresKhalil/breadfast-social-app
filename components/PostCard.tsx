import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { styled } from 'nativewind';
import UserAvatar from './UserAvatar';

interface Post {
  id: number;
  user_id: number;
  title: string;
  body: string;
}

interface PostCardProps {
  post: Post;
  onPress: () => void;
}

const StyledTouchableOpacity = styled(TouchableOpacity);
const StyledView = styled(View);
const StyledText = styled(Text);

const PostCard: React.FC<PostCardProps> = ({ post, onPress }) => {
  return (
    <StyledTouchableOpacity
      className="bg-white p-4 mb-4 rounded-lg shadow-lg hover:bg-gray-100 active:bg-gray-200 transition-colors duration-200"
      onPress={onPress}
    >
      <StyledView className="flex-row items-center mb-2 p-2">
        <UserAvatar userId={post.user_id} />
        <StyledView className="ml-3">
          <StyledText className="font-bold text-lg text-gray-900">
            {post.title}
          </StyledText>
          <StyledText className="text-sm text-gray-500">
            by User {post.user_id}
          </StyledText>
        </StyledView>
      </StyledView>
      <StyledText className="text-gray-700 leading-relaxed">
        {post.body}
      </StyledText>
    </StyledTouchableOpacity>
  );
};

export default PostCard;
