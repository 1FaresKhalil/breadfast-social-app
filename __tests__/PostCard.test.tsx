import React from 'react';
import renderer from 'react-test-renderer';
import PostCard from '../components/PostCard';

const post = {
  id: 1,
  user_id: 1,
  title: 'Test Post',
  body: 'This is a test post body.',
};

describe('<PostCard />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PostCard post={post} onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
