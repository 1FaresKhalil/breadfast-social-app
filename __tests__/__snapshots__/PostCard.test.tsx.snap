// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "shadowColor": "rgba(0, 0, 0, 0.1)",
      "shadowOffset": {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 15,
    }
  }
>
  <View
    style={
      [
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "marginBottom": 8,
          },
          {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          [
            {
              "width": 40,
            },
            {
              "height": 40,
            },
            {
              "backgroundColor": "#d1d5db",
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "overflow": "hidden",
            },
          ],
        ]
      }
    >
      <Image
        source={
          {
            "uri": "https://robohash.org/1.png",
          }
        }
        style={
          [
            [
              {
                "width": "100%",
              },
              {
                "height": "100%",
              },
            ],
          ]
        }
      />
    </View>
    <View
      style={
        [
          [
            {
              "marginLeft": 12,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            [
              {
                "fontWeight": "700",
              },
              {
                "fontSize": 18,
                "lineHeight": 28,
              },
              {
                "color": "#111827",
              },
            ],
          ]
        }
      >
        Test Post
      </Text>
      <Text
        style={
          [
            [
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#6b7280",
              },
            ],
          ]
        }
      >
        by User 
        1
      </Text>
    </View>
  </View>
  <Text
    style={
      [
        [
          {
            "color": "#374151",
          },
        ],
      ]
    }
  >
    This is a test post body.
  </Text>
</View>
`;
